### PR TITLE
feat: implement app/environment admin operations, make GraphQL operations/inputs consistent

### LINKED ISSUE #242: Add option for developer to add new `AppEnv` for their `App`
Follow-up from #181

### PR DESCRIPTION
**Pull request by beeman**
Closes  #242

### CODE REVIEW

File: libs/api/app/data-access/src/lib/api-app-user-data-access.service.ts [hunk 1]
```diff
@@ -130,12 +135,62 @@ export class ApiAppUserDataAccessService {
...
   }
 
-  async userUpdateApp(userId: string, appId: string, data: AppUpdateInput) {
+  async userUpdateApp(userId: string, appId: string, data: UserAppUpdateInput) {
     await this.data.ensureAppOwner(userId, appId)
     return this.data.app.update({ where: { id: appId }, data, include: this.app.include })
   }
 
-  async userUpdateAppEnv(userId: string, appId: string, appEnvId: string, data: AppEnvUpdateInput) {
+  async userCreateAppEnv(userId: string, appId: string, clusterId: string, data: UserAppEnvCreateInput) {
```
Comment (beeman) [inline libs/api/app/data-access/src/lib/api-app-user-data-access.service.ts:143]: This method allows the user to create new environments in a cluster.

File: libs/api/app/data-access/src/lib/dto/user-app-transaction-list.input.ts [hunk 2]
```diff
@@ -2,10 +2,12 @@ import { Field, InputType } from '@nestjs/graphql'
 import { AppTransactionStatus } from '../entity/app-transaction-status.enum'
 
 @InputType()
-export class AppTransactionListInput {
+export class UserAppTransactionListInput {
   @Field({ nullable: true })
   destination?: string
   @Field({ nullable: true })
+  ip?: string
+  @Field({ nullable: true })
```
Comment (beeman) [inline libs/api/app/data-access/src/lib/dto/user-app-transaction-list.input.ts:10]: We can now filter AppTransactions on `ip` and `ua`

File: libs/api/app/feature/src/lib/api-app-admin-feature.resolver.ts [hunk 3]
```diff
@@ -9,10 +14,15 @@ export class ApiAppAdminFeatureResolver {
   constructor(private readonly service: ApiAppAdminDataAccessService) {}
 
   @Mutation(() => App, { nullable: true })
-  adminCreateApp(@CtxUser() user: User, @Args('input') input: AppCreateInput) {
+  adminCreateApp(@CtxUser() user: User, @Args('input') input: AdminAppCreateInput) {
     return this.service.adminCreateApp(user.id, input)
   }
 
+  @Mutation(() => App, { nullable: true })
+  adminUpdateApp(@CtxUser() user: User, @Args('appId') appId: string, @Args('input') input: AdminAppUpdateInput) {
```
Comment (beeman) [inline libs/api/app/feature/src/lib/api-app-admin-feature.resolver.ts:22]: Allow Admin to update an app.

File: libs/api/app/feature/src/lib/api-app-env-admin-feature.resolver.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+import { User } from '@kin-kinetic/api/user/data-access'
+import { UseGuards } from '@nestjs/common'
+import { Args, Mutation } from '@nestjs/graphql'
+
+@UseGuards(ApiAuthGraphqlGuard)
+export class ApiAppEnvAdminFeatureResolver {
+  constructor(private readonly service: ApiAppEnvAdminDataAccessService) {}
+
+  @Mutation(() => AppEnv, { nullable: true })
+  adminDeleteAppEnv(@CtxUser() user: User, @Args('appId') appId: string, @Args('appEnvId') appEnvId: string) {
```
Comment (beeman) [inline libs/api/app/feature/src/lib/api-app-env-admin-feature.resolver.ts:12]: Allow Admin to delete an app env.

File: libs/api/cluster/data-access/src/lib/api-cluster-user-data-access.service.ts [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import { ApiCoreDataAccessService } from '@kin-kinetic/api/core/data-access'
+import { Injectable } from '@nestjs/common'
+
+@Injectable()
+export class ApiClusterUserDataAccessService {
+  constructor(private readonly data: ApiCoreDataAccessService) {}
+
+  async userClusters(userId: string) {
```
Comment (beeman) [inline libs/api/cluster/data-access/src/lib/api-cluster-user-data-access.service.ts:8]: Allow user to list clusters

File: libs/api/app/data-access/src/lib/api-app-env-admin-data-access.service.ts [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+@Injectable()
+export class ApiAppEnvAdminDataAccessService {
+  private readonly logger = new Logger(ApiAppEnvAdminDataAccessService.name)
+
+  constructor(
+    private readonly appAdmin: ApiAppAdminDataAccessService,
+    private readonly data: ApiCoreDataAccessService,
+  ) {}
+
+  async adminDeleteAppEnv(userId: string, appId: string, appEnvId: string) {
```
Comment (beeman) [inline libs/api/app/data-access/src/lib/api-app-env-admin-data-access.service.ts:14]: On deletion of the App Env, it first need to remove all the related data.

**Review by beeman — COMMENTED** [review]
The bulk of this PR consist of renames to improve consistency in our GraphQL inputs (things like `AppCreateInput` -> `AdminAppCreateInput`).

I've added a comment in the actual admin operations that got added.

**Review by RamirezAlex — APPROVED** [review]

**Pull request merged**